### PR TITLE
Hotfix/eq invite check

### PR DESCRIPTION
**Pull request by SavoX2**

### CODE REVIEW

File: routes.js [hunk 1]
```diff
@@ -40,25 +40,27 @@ router.get('/user-type', auth.authUser, (req, res) => userTypeController.getAll(
 router.get('/user-type/:id', auth.authUser, (req, res) => userTypeController.getById(req, res));
 
 // Company routes
-router.post('/company', auth.authAdmin, (req, res) => companyController.create(req, res));
-router.get('/company', auth.authAdmin, (req, res) => companyController.getAll(req, res));
-router.get('/company/:id', auth.authAdmin, (req, res) => companyController.getById(req, res));
+router.post('/company', (req, res) => companyController.create(req, res));
```
Comment (srkacha) [inline routes.js]: Why did you remove the auth middleware? I thought all of our api should be protected with authentication.

File: routes.js [hunk 2]
```diff
@@ -40,25 +40,27 @@ router.get('/user-type', auth.authUser, (req, res) => userTypeController.getAll(
...
-router.get('/company/get-by-name/:name', auth.authAdmin, (req, res) => companyController.getByNameContains(req, res));
+router.delete('/company/:id', (req, res) => companyController.deleteCompany(req, res));
+router.get('/company/get-by-name/:name', (req, res) => companyController.getByNameContains(req, res));
 
 // EventQuestion routes 
 router.post('/event-question', auth.authUser, (req, res) => eventQuestionController.create(req, res));
-router.get('/event-question', auth.authUser, (req, res) => eventQuestionController.getAll(req, res));
+// this is kinda dumb, but the way it was set up before you were getting every event question for every event 
+// in the company, it is only natural that you would get only the ones for a specific event
+router.get('/event-question/all/:id', auth.authUser, (req, res) => eventQuestionController.getAll(req, res));
```
Comment (srkacha) [inline routes.js]: Change this 'id' to 'event_id' to make the path more understandable. 'id' alone would make sense if it was in relation to event question, but it is related to event. Same for other paths.

File: controller/eventQuestionController.js [hunk 3]
```diff
@@ -153,49 +213,53 @@ const addResponseToEvent = (req, res) => {
...
     };
 
-    // allow commenting only on company events; TODO: additional check if user has accepted invite for event (after test data and controller for invite are made)
-    EventQuestion.findOneAndUpdate({ _id: new ObjectId(eventQuestionId), companyID: user.companyID }, { $push: { responses: response } }, { useFindAndModify: false, new: true, runValidators: true }).then(eventQuestion => {
-        if (!eventQuestion) {
-            return res.status(HttpStatus.NOT_FOUND).json({
-                status: StatusEnum['ERROR'],
-                message: 'Event question not found'
+    if (ValidationUtil.isUserAdmin(user.accessType)) {
+        return addResponseToEventHelper(req, res, new ObjectId(eventQuestionID), user.companyID, response);
```
Comment (srkacha) [inline controller/eventQuestionController.js]: Moving parts of code to helper functions makes sense if you're going to reuse the function. This way it only makes us think about two functions at once. @zirlman @djuricic1 your thoughts would be welcome on this

File: routes.js (hunk 1 above)
Comment (SavoX2) [reply routes.js]: I wrote comments in the code for as to why. At the moment, all companies are created by us. There is no use case in which an `admin`, `super-admin` level user should be able to create a new company. Every company created is assigned to a super-admin type user who then on creates admins and employees for that company. Until we agree to create a new user-type for us: ex. Super-Super-Admin I left out authentication methods.

File: controller/eventQuestionController.js (hunk 3 above)
Comment (SavoX2) [reply controller/eventQuestionController.js]: This way it only makes us think about two functions at once. - Elaborate.

File: routes.js (hunk 1 above)
Comment (SavoX2) [reply routes.js]: This also extends to that no super-admin should be allowed to delete a company.

File: controller/eventQuestionController.js (hunk 3 above)
Comment (srkacha) [reply controller/eventQuestionController.js]: What I meant to say is, the flow of the function is easier to follow if it is just a continuous piece of code, when it is split in two functions I have to keep track of the old variables and the new ones, what was passed to the function, what was validated outside, and what inside of the new one and stuff like that. I guess it's a thing of coding preference, so leave it at that.

Comment (SavoX2) [reply controller/eventQuestionController.js]: I don't like to have duplicate code if I don't need to. Helper functions help minimize that amount. Also it is not that some complicated objects were passed, helper functions usually take simple parameters like these ones. I think it keeps the code cleaner and more maintainable.

File: routes.js (hunk 1 above)
Comment (SavoX2) [reply routes.js]: I realize that the way it is now, anyone can create or delete a company, but until a new user-type is agreed upon, I will leave it at this.

File: controller/eventQuestionController.js [hunk 4]
```diff
@@ -24,119 +25,178 @@ const create = (req, res) => {
...
             message: 'Bad request'
         });
     }
 
-    EventQuestion.findOne({ _id: new ObjectId(eventQuestionId), companyID: new ObjectId(user.companyID) }, { __v: 0 }).then(eventQuestion => {
-        if (!eventQuestion) {
-            return res.status(HttpStatus.NOT_FOUND).json({
-                status: StatusEnum['ERROR'],
-                message: 'Event question not found'
+    if (ValidationUtil.isUserAdmin(user.accessType)) {
```
Comment (zirlman) [inline controller/eventQuestionController.js]: handled by auth mw

File: controller/eventQuestionController.js [hunk 5]
```diff
@@ -24,119 +25,178 @@ const create = (req, res) => {
...
         return res.status(HttpStatus.BAD_REQUEST).json({
-            status: StatusEnum['ERROR'],
             message: 'Bad request'
         });
     }
 
-    EventQuestion.find({ companyID: user.companyID }).then(eventQuestions => {
-        return res.status(HttpStatus.OK).json(eventQuestions);
+    // if it is an admin level user, he can get all questions regardless of invite or not => GOOD, BAD? Discuss
+    if (ValidationUtil.isUserAdmin(user.accessType)) {
```
Comment (zirlman) [inline controller/eventQuestionController.js]: handled by auth mw

Comment (SavoX2) [reply controller/eventQuestionController.js]: Refer to comment above the `if` line.

File: controller/eventQuestionController.js (hunk 4 above)
Comment (SavoX2) [reply controller/eventQuestionController.js]: Again, same as previous comment above.
